ci_framework/roles/libvirt_manager/templates/domain.xml.j2

```diff
@@ -0,0 +1,106 @@
+<domain type='kvm'>
+  <name>cifmw-{{ vm_type }}-{{ vm_id }}</name>
+  <memory unit='GB'>{{ vm_data.value.memory | default(2) }}</memory>
+  <vcpu placement='static'>{{ vm_data.value.cpu | default(2) }}</vcpu>
+  <os>
+    <type arch='x86_64' machine='q35'>hvm</type>
+    <boot dev='hd'/>
+    <bootmenu enable='no'/>
+  </os>
+  <cpu mode='host-passthrough' check='none'/>
+  <clock offset='utc'>
+    <timer name='rtc' tickpolicy='catchup'/>
+    <timer name='pit' tickpolicy='delay'/>
+    <timer name='hpet' present='no'/>
+  </clock>
+  <features>
+    <acpi/>
+    <apic/>
+  </features>
+  <devices>
+    <emulator>/usr/libexec/qemu-kvm</emulator>
+    <disk type='file' device='disk'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='{{ cifmw_reproducer_basedir }}/workload/{{ vm_type }}-{{ vm_id }}.qcow2'/>
+      <target dev='sda' bus='sata'/>
+    </disk>
+    {% for network in vm_data.value.nets %}
+    <interface type='network'>
+      <source network='{{ network }}'/>
+      <mac address='{{ '24:42:%02i'|format(vm_id|int) |random_mac }}'/>
```

Comment:
The offset in the nets might be one octet in the mac.

Reply:
in that case we have some more randomness, not a big deal. We ensure we also have enough difference using the vm_id/index, there are less chances we'll hit any collision.
In the past, in tripleo-lab, I *had* collisions before using the VM index as a part of the MAC.